pkg/services/grafana.go

```diff
@@ -17,9 +17,13 @@ import (
 )
 
 type GrafanaOptions struct {
-	ApiUrl             string `json:"apiUrl"`
-	ApiKey             string `json:"apiKey"`
-	InsecureSkipVerify bool   `json:"insecureSkipVerify"`
+	ApiUrl              string        `json:"apiUrl"`
+	ApiKey              string        `json:"apiKey"`
+	InsecureSkipVerify  bool          `json:"insecureSkipVerify"`
+	MaxIdleConns        int           `json:"maxIdleConns"`
+	MaxIdleConnsPerHost int           `json:"maxIdleConnsPerHost"`
+	MaxConnsPerHost     int           `json:"maxConnsPerHost"`
+	IdleConnTimeout     time.Duration `json:"idleConnTimeout"`
```

Comment:
We can't use time.Duration for parsing time values - we will need to use a string and then parse the actual `time.Duration` by doing something like this 

```golang
	idleConnTimeout, err := time.ParseDuration(s.opts.IdleConnTimeout)
	if err != nil {
		return errors.Wrap(err, "failed to parse idle connection timeout")
	}
```

and then passing in `idleConnTimeout` to the `NewTransport` function.